app/mqtt.mjs

```diff
@@ -56,6 +43,43 @@ export const publishValues = async (modbusClient, mqttClient) => {
         topicMap[topicName] = JSON.stringify(value)
     }
 
+    await publishTopics(mqttClient, topicMap)
+}
+
+const publishFlags = async (modbusClient, mqttClient) => {
+    // Create a map from topic name to value that should be published
+    let topicMap = {}
+
+    // Publish state for each mode.
+    const modeSummary = await getFlagSummary(modbusClient)
+
+    for (const [mode, state] of Object.entries(modeSummary)) {
+        const topicName = `${TOPIC_PREFIX_MODE}/${mode}`
+
+        // Boolean values are changed to "ON" and "OFF" respectively since those are the
+        // defaults for MQTT switches in Home Assistant
+        topicMap[topicName] = state ? 'ON' : 'OFF'
+    }
+
+    await publishTopics(mqttClient, topicMap)
+}
+
+const publishSettings = async (modbusClient, mqttClient, settingName = '') => {
+    // Create a map from topic name to value that should be published
+    let topicMap = {}
+    const settings = await getSettings(modbusClient)
+
+    for (const [setting, value] of Object.entries(settings)) {
+        if(settingName === '' || settingName == setting) {
```

Comment:
I'm not sure this check is worth it - we're requesting all settings from Modbus anyway so this won't make anything faster AFAICT? Might as well keep it simple.

Reply:
True, I´ll change